src/api/githubService.js

```diff
@@ -0,0 +1,284 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import axios from "axios";
+import moment from "moment";
+
+const functionalTestsWorkflowId = 5937682;
+
+export const getAllWorkflows = async () => {
+  const api = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/actions/runs?branch=main`;
+
+  let wfRunsInfo = [];
+  let workflows = {};
+
+  if (await _isWithinExpiryTime("awf_expireTime"))
+    return await _getDataFromLocal("all_workflows");
+
+  const {
+    data: { workflow_runs },
+  } = await axios.get(api);
+
+  for (let workflow_run of workflow_runs) {
+    if (workflows[workflow_run.name] === undefined) {
+      workflows[workflow_run.name] = workflow_run.id;
```

Comment:
in this way you take just the first workflow with this name. Is this what we want? What about if the same workflow has been executed multiple times?

Reply:
what i want to achieve here is get the latest build for all our workflows